addons/l10n_anz_ubl_pint/models/account_edi_xml_pint_anz.py

```diff
@@ -0,0 +1,109 @@
+        vals_list = super()._get_invoice_tax_totals_vals_list(invoice, taxes_vals)
+        company_currency = invoice.company_id.currency_id
+        if invoice.currency_id != company_currency:
+            # if company currency != invoice currency, need to add a TaxTotal section
+            # see https://docs.peppol.eu/poac/aunz/pint-aunz/bis/#_tax_in_accounting_currency
+            tax_totals_vals = {
+                'currency': company_currency,
+                'currency_dp': company_currency.decimal_places,
+                'tax_amount': taxes_vals['tax_amount'],
+                'tax_subtotal_vals': [],
+            }
+            for _grouping_key, vals in taxes_vals['tax_details'].items():
+                tax_totals_vals['tax_subtotal_vals'].append({
+                    'currency': company_currency,
+                    'currency_dp': company_currency.decimal_places,
+                    'taxable_amount': vals['base_amount'],
+                    'tax_amount': vals['tax_amount'],
+                    'percent': vals['_tax_category_vals_']['percent'],
+                    'tax_category_vals': vals['_tax_category_vals_'],
+                })
+            vals_list.append(tax_totals_vals)
+        return vals_list
+
+    def _get_tax_unece_codes(self, invoice, tax):
+        """ The GST category must be provided in the file.
+        See https://docs.peppol.eu/poac/aunz/pint-aunz/bis/#_tax_category_code
+        """
+        # OVERRIDE account_edi_ubl_cii
+        supplier = invoice.company_id.partner_id.commercial_partner_id
+        if not supplier.vat:
+            return 'O'  # a business is not registered for GST, the business cannot issue tax invoices. In this case, the GST category code should be O (Outside scope of tax).
+        return super()._get_tax_unece_codes(invoice, tax)
+
+    def _get_tax_category_list(self, invoice, taxes):
+        # EXTENDS account_edi_ubl_cii
+        vals_list = super()._get_tax_category_list(invoice, taxes)
+        for vals in vals_list:
+            vals['tax_scheme_vals'] = {'id': 'GST'}
+        return vals_list
+
+    def _get_customization_ids(self):
+        vals = super()._get_customization_ids()
+        vals['pint_anz'] = 'urn:peppol:pint:billing-1@aunz-1'
+        return vals
+
+    def _get_partner_party_legal_entity_vals_list(self, partner):
+        # EXTENDS account.edi.xml.ubl_bis3
+        vals_list = super()._get_partner_party_legal_entity_vals_list(partner)
+
+        for vals in vals_list:
+            if partner.country_code == 'AU' and partner.vat:
+                vals.update({
+                    'company_id': partner.vat.replace(" ", ""),
+                    'company_id_attrs': {'schemeID': '0151'},
+                })
+            if partner.country_code == 'NZ':
+                vals.update({
+                    'company_id': partner.company_registry,
+                    'company_id_attrs': {'schemeID': '0088'},
+                })
```

Comment:
don't forget to adapt this during the fw-port (in 17.0, we use the peppol_endpoint and peppol_eas). We may probably even be able to remove this function.

Reply:
This is in 17 :thinking: 
I think I'll remove it and use the peppol fields. We can add something back later if needed.
In theory the spec says we have to use a scheme from the eas list so it should be enough.